### PR TITLE
[IR] Allow llvm.ptrmask of vectors

### PR DESCRIPTION
**Pull request by nikic**
llvm.ptrmask is currently limited to pointers only, and does not accept vectors of pointers. This is an unnecessary limitation, especially as the underlying instructions (getelementptr etc) do support vectors of pointers.

We should relax this sooner rather than later, to avoid introducing code that assumes non-vectors (#67166).

### CODE REVIEW

**Comment by goldsteinn** [comment]
Will this become an issue: https://reviews.llvm.org/D153773#inline-1486422
?

**Comment by nikic** [comment]
> Will this become an issue: https://reviews.llvm.org/D153773#inline-1486422 ?

No, the final implementation used getScalarSizeInBits(): https://github.com/llvm/llvm-project/blob/50abfc4298980fc219c7d70ca2567c61e856b614/llvm/lib/Analysis/ValueTracking.cpp#L1636

File: llvm/docs/LangRef.rst [hunk 1]
```diff
@@ -26865,7 +26865,8 @@ Syntax:
 Arguments:
 """"""""""
 
-The first argument is a pointer. The second argument is an integer.
+The first argument is a pointer or vector of pointers. The second argument is
+an integer or vector of integers.
```
Comment (fhahn) [inline llvm/docs/LangRef.rst:26869]: nit: the working below still refers to a single pointer, maybe adjust to something like `pointer(s)`?

**Review by fhahn — APPROVED** [review]
LGTM, thanks! 

I don't think there was any particular reason for limiting this to pointers only.

**Pull request merged**